packaging_automation/citus_package.py

```diff
@@ -0,0 +1,263 @@
+        raise ValueError("Platforms should have two parts divided by '/' or should be 'pgxn' ")
+    elif len(parts) == 1 and parts[0] == "pgxn":
+        os_name = "pgxn"
+        os_release = ""
+    else:
+        os_name = parts[0]
+        os_release = parts[1]
+        if os_name not in supported_platforms:
+            raise ValueError(
+                f"{os_name} is not among supported operating systems. Supported operating systems are as below:\n "
+                f"{','.join(supported_platforms.keys())}")
+        if os_release not in supported_platforms[os_name]:
+            raise ValueError(f"{os_release} is not among supported releases for {os_name}."
+                             f"Supported releases are as below:\n {','.join(supported_platforms[os_name])}")
+    return os_name, os_release
+
+
+def is_docker_running() -> bool:
+    try:
+        docker_client = docker.from_env()
+        docker_client.ping()
+        return True
+    except:
+        return False
+
+
+@validate_parameters
+def get_signing_credentials(packaging_secret_key: str,
+                            packaging_passphrase: non_empty(non_blank(str))) -> Tuple[str, str]:
+    if packaging_secret_key:
+        secret_key = packaging_secret_key
+    else:
+        fingerprints = get_gpg_fingerprints_by_name(GPG_KEY_NAME)
+        if len(fingerprints) == 0:
+            raise ValueError(f"Key for {GPG_KEY_NAME} does not exist")
+
+        gpg = gnupg.GPG()
+
+        private_key = gpg.export_keys(fingerprints[0], secret=True, passphrase=packaging_passphrase)
+        secret_key = transform_key_into_base64_str(private_key)
+
+    passphrase = packaging_passphrase
+    return secret_key, passphrase
+
+
+def sign_packages(base_output_path: str, sub_folder: str, secret_key: str, passphrase: str, input_files_dir: str,
+                  output_validation: bool = False):
+    output_path = f"{base_output_path}/{sub_folder}"
+    deb_files = glob.glob(f"{output_path}/*.deb", recursive=True)
+    rpm_files = glob.glob(f"{output_path}/*.rpm", recursive=True)
+    os.environ["PACKAGING_PASSPHRASE"] = passphrase
+    os.environ["PACKAGING_SECRET_KEY"] = secret_key
+
+    if len(rpm_files) > 0:
+        print("Started RPM Signing...")
+        result = run_with_output(f"docker run --rm -v {output_path}:/packages/{sub_folder} -e PACKAGING_SECRET_KEY -e "
+                                 f"PACKAGING_PASSPHRASE citusdata/packaging:rpmsigner", text=True)
+        output = result.stdout
+        print(f"Result:{output}")
+        if result.returncode != 0:
+            raise ValueError(f"Error while signing rpm files.Err:{result.stderr}")
+        if output_validation:
+            validate_output(output, f"{input_files_dir}/packaging_ignore.yml", PackageType.rpm)
+
+        print("RPM signing finished successfully.")
+
+    if len(deb_files) > 0:
+        print("Started DEB Signing...")
+
+        result = subprocess.run(
```

Comment:
Any particular reason why we do not use `run_with_output` here?

Reply:
input is requested. run_with_output is not designed to use with input